Editor/EditorCore/MenuAction.cs

```diff
@@ -268,21 +280,26 @@ public ActionResult PerformAction()
         }
 
         /// <summary>
-        /// Perform whatever action this menu item is supposed to do. This method should never been call directly
-        /// but though PerformAction.
+        /// Performs the action for this menu item. Use this method to implement the action and then
+        /// use PerformAction to call it.
         /// </summary>
         /// <returns>A new ActionResult with a summary of the state of the action's success.</returns>
         protected abstract ActionResult PerformActionImplementation();
 
+        const string obsoleteDoActionMsg = "DoAction() has been replaced by PerformAction(), the implementation of the action should inherits from PerformActionImplementation(). (UnityUpgradable) -> PerformAction()";
         /// <summary>
         /// Perform whatever action this menu item is supposed to do. You are responsible for implementing Undo.
```

Comment:
So what I hear you saying is that the dev implementing a new action isn't managing the call to the Undo stack in `ActionResult.PerformAction`, but rather in `MyNewAction.PerformAction`. In that case, it's weird to act like the reader would be implementing the `ActionResult.PerformAction` function. Is that right?

Reply:
Yeah, Actually a dev creating a new menu action `MyMenuAction` will have to implement the `PerformAction()` method. In that method, the user will have to him/herself register the modified object to the Undo stack (using for instance `Undo.RecordObject(myObject)` or any other applicable `Undo.Register...()` method ) so that using undo in Unity would revert the changes done by the user in `MyMenuAction.PerformAction`

Reply:
I updated the one on PerformAction and removed the rest